packages/reactive-element/src/decorators/custom-element.ts

```diff
@@ -4,4 +4,59 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-export * from '../legacy-decorators/custom-element.js';
+/*
+ * IMPORTANT: For compatibility with tsickle and the Closure JS compiler, all
+ * property decorators (but not class decorators) in this file that have
+ * an @ExportDecoratedItems annotation must be defined as a regular function,
+ * not an arrow function.
+ */
+
+import type {Constructor} from './base.js';
+
+/**
+ * Allow for custom element classes with private constructors
+ */
+type CustomElementClass = Omit<typeof HTMLElement, 'new'>;
+
+export type CustomElementDecorator = {
+  // legacy
+  (cls: CustomElementClass): void;
+
+  // standard
+  (
+    target: CustomElementClass,
+    context: ClassDecoratorContext<Constructor<HTMLElement>>
+  ): void;
+};
+
+/**
+ * Class decorator factory that defines the decorated class as a custom element.
+ *
+ * ```js
+ * @customElement('my-element')
+ * class MyElement extends LitElement {
+ *   render() {
+ *     return html``;
+ *   }
+ * }
+ * ```
+ * @category Decorator
+ * @param tagName The tag name of the custom element to define.
+ */
+export const customElement =
+  (tagName: string): CustomElementDecorator =>
+  (
+    classOrTarget: CustomElementClass | Constructor<HTMLElement>,
+    context?: ClassDecoratorContext<Constructor<HTMLElement>>
+  ) => {
+    if (context !== undefined) {
+      context.addInitializer(() => {
```

Comment:
Why not always immediately call customElements.define? What happens as part of class finalization that it might care about?

I guess another decorator could mess with the prototype chain, add custom element callbacks…

Ok, yeah, I see

Reply:
I tried to add a test for this and failed for now, but addInitializer() should go after static fields are assigned, whereas accessing the class immediately would be before that. I tried to make some test that triggered that, but a static getters (I used it for `observedAttributes`) are installed early. Anyway, I know this is the right timing, we can make a test later?